### PR TITLE
support numpy dtype and polish code of list index.

### PR DESCRIPTION
**Pull request by hbwx24**
### PR types

Bug fixes
### PR changes

APIs
### Describe

- `varbase._getitem__` 支持 np.longlong，np.int64，np.int32 ，np.int16
- 修改 list索引的判断条件。动态图这种联合索引仍然走numpy分支
  ```python
  tensor_x[:, tensor_y1:tensor_y2] = 42
  ```

### CODE REVIEW

File: python/paddle/fluid/dygraph/varbase_patch_methods.py [hunk 1]
```diff
@@ -588,18 +588,39 @@ def _is_list_tuple(item):
...
-            if not isinstance(item, tuple):
+            if not isinstance(item, (tuple, list)):
                 item = [item]
 
             for slice_item in item:
-                if isinstance(slice_item, list):
+                if isinstance(slice_item, (list, tuple)):
                     return True
                 elif isinstance(slice_item, Variable):
                     return True
```
Comment (zyfncg) [inline python/paddle/fluid/dygraph/varbase_patch_methods.py]: when item is a list, the return is False

Comment (hbwx24) [reply python/paddle/fluid/dygraph/varbase_patch_methods.py]: done, thx.

**Review by zyfncg — COMMENTED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**